app/controllers/submissions_controller.rb

```diff
@@ -20,14 +20,19 @@ class SubmissionsController < ApplicationController
                          :repo_browser,
                          :set_result_marking_state,
                          :update_submissions,
+                         :collect_submissions,
+                         :run_tests,
                          :populate_submissions_table,
                          :populate_peer_submissions_table]
   before_action :authorize_for_ta_and_admin,
                 only: [:index,
                        :browse,
                        :manually_collect_and_begin_grading,
+                       :set_result_marking_state,
                        :revisions,
                        :repo_browser,
+                       :collect_submissions,
```

Comment:
No need to add this here, since it is handled by the `authorize!` call down below. 

In fact, it seems to me like you should also add `run_tests` and `manually_collect_and_begin_grading` to the line 44 list, and remove them from this `only` list.